### PR TITLE
mail-notification: init at 5.4

### PR DESCRIPTION
**Pull request by proteansec**
###### Motivation for this change

Addition of mail-notification packages.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/desktops/gnome-2/desktop/mail-notification/default.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    # copy executable
+    cp ./build/src/mail-notification $out/bin/
+
+    # copy icons
+    #mkdir -p $out/share/icons/hicolor
+    cp -r art/* $out/share/
+
+    # copy schemas
+    mkdir -p $out/etc/gconf/schemas/
+    #GCONF_CONFIG_SOURCE="xml:merged:${gnome2.GConf}/etc/gconf/gconf.xml.defaults" ${gnome2.GConf}/bin/gconftool-2 --makefile-install-rule data/mail-notification.schemas.in.in
```
Comment (zimbatm) [inline pkgs/desktops/gnome-2/desktop/mail-notification/default.nix]: is this a debug leftover?

File: pkgs/desktops/gnome-2/desktop/mail-notification/default.nix [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    inherit rev;
+    owner = "epienbroek";
+    repo = "mail-notification";
+    sha256 = "1slb7gajn30vdaq0hf5rikwdly1npmg1cf83hpjs82xd98knl13d";
+  };
+
+   buildInputs = [ pkgconfig glib dbus_glib gmime libnotify libgnome_keyring openssl cyrus_sasl gnonlin sylpheed gob2 gettext intltool gnome2.GConf gnome2.libgnomeui dbus_glib gmime libnotify gnome2.gnome_keyring gnome2.scrollkeeper libxml2 gnome2.gnome_icon_theme hicolor_icon_theme tango-icon-theme ];
+
+  prePatch = ''
+    sed -i  -e '/jb_rule_set_install_message/d' -e '/jb_rule_add_install_command/d' jbsrc/jb.c || die
```
Comment (joachifm) [inline pkgs/desktops/gnome-2/desktop/mail-notification/default.nix]: Where does `die()` come from? I'd expect `set -e` in the generic setup script would be sufficient, is it not?

File: pkgs/desktops/gnome-2/desktop/mail-notification/default.nix [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    # copy properties dialogs
+    cp ui/* $out/share/
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    description = "Tray status icon, which notifies us when new email arrives";
+    homepage = "http://www.nongnu.org/mailnotify/";
+    license = "GPL-3";
```
Comment (zimbatm) [inline pkgs/desktops/gnome-2/desktop/mail-notification/default.nix]: that should be `lib.licenses.gpl3`

**Comment by joachifm** [comment]
The patches are too large, please fetch from an external source if possible.

File: pkgs/desktops/gnome-2/desktop/mail-notification/default.nix (hunk 2 above)
Comment (proteansec) [inline pkgs/desktops/gnome-2/desktop/mail-notification/default.nix]: fixed

File: pkgs/desktops/gnome-2/desktop/mail-notification/default.nix (hunk 1 above)
Comment (proteansec) [inline pkgs/desktops/gnome-2/desktop/mail-notification/default.nix]: yeah, it's a debug leftover, which was now removed

File: pkgs/desktops/gnome-2/desktop/mail-notification/default.nix [hunk 4]
```diff
@@ -15,17 +15,16 @@ stdenv.mkDerivation rec {
    buildInputs = [ pkgconfig glib dbus_glib gmime libnotify libgnome_keyring openssl cyrus_sasl gnonlin sylpheed gob2 gettext intltool gnome2.GConf gnome2.libgnomeui dbus_glib gmime libnotify gnome2.gnome_keyring gnome2.scrollkeeper libxml2 gnome2.gnome_icon_theme hicolor_icon_theme tango-icon-theme ];
 
   prePatch = ''
-    sed -i  -e '/jb_rule_set_install_message/d' -e '/jb_rule_add_install_command/d' jbsrc/jb.c || die
+    set -e
```
Comment (Mic92) [inline pkgs/desktops/gnome-2/desktop/mail-notification/default.nix]: `set -e` is already enabled by default

Comment (proteansec) [inline pkgs/desktops/gnome-2/desktop/mail-notification/default.nix]: Fixed.

**Comment by Mic92** [comment]
I simplified the build:

```diff
diff --git a/pkgs/desktops/gnome-2/desktop/mail-notification/default.nix b/pkgs/desktops/gnome-2/desktop/mail-notification/default.nix
index 05c2384afe..36608f4294 100644
--- a/pkgs/desktops/gnome-2/desktop/mail-notification/default.nix
+++ b/pkgs/desktops/gnome-2/desktop/mail-notification/default.nix
@@ -17,57 +17,29 @@ stdenv.mkDerivation rec {
   prePatch = ''
     sed -i  -e '/jb_rule_set_install_message/d' -e '/jb_rule_add_install_command/d' jbsrc/jb.c
 
-    # Ensure we never append -Werror
-    sed -i -e 's/ -Werror//' jb jbsrc/jb.c
+    # currently disable the check for missing sheme until a better solution
+    # is found; needed, because otherwise the application doesn't even start
+    # and fails saying it unable to find gconf scheme values.
+    sed -i -e 's/(schema_missing)/(!schema_missing)/g' src/mn-conf.c
   '';
 
   patches = [
-    ./patches/mail-notification-5.4-remove-ubuntu-special-case.patch
     ./patches/mail-notification-dont-link-against-bsd-compat.patch
   ];
 
   patchFlags = "-p0";
+  NIX_CFLAGS_COMPILE = "-Wno-error";
 
-  configurePhase = ''
-    ./jb configure
-    runHook postConfigure
-  '';
+  preConfigure = "./jb configure prefix=$out";
 
   postConfigure = ''
-    sed -i -e "/^prefix/c prefix|string|1|$out" \
-           -e 's/^pkgdatadir.*/pkgdatadir|string|0|$datadir/g' \
-           build/config
-
-    # currently disable the check for missing sheme until a better solution
-    # is found; needed, because otherwise the application doesn't even start
-    # and fails saying it unable to find gconf scheme values.
-    sed -i -e 's/(schema_missing)/(!schema_missing)/g' src/mn-conf.c
+    substituteInPlace build/config \
+      --replace "omf-dir|string|1|${gnome2.scrollkeeper}/share/omf" "omf-dir|string|1|$out/share/omf" \
+      --replace "scrollkeeper-dir|string|1|${gnome2.scrollkeeper}/var/lib/scrollkeeper" "omf-dir|string|1|$out/var/lib/scrollkeeper" \
   '';
 
-
-  buildPhase = ''
-    ./jb build
-  '';
-
-  # the "./jb install" tries to install directly to nix store, which is not
-  # writable, so we have to copy the files ourselves
-  installPhase = ''
-    # create directories
-    mkdir -p $out/{bin,lib,libexec,share,etc,var}
-
-    # copy executable
-    cp ./build/src/mail-notification $out/bin/
-
-    # copy icons
-    cp -r art/* $out/share/
-
-    # copy schemas
-    mkdir -p $out/etc/gconf/schemas/
-    cp data/mail-notification.schemas.in.in $out/etc/gconf/schemas/
-
-    # copy properties dialogs
-    cp ui/* $out/share/
-  '';
+  buildPhase = "./jb build";
+  installPhase = "./jb install";
 
   enableParallelBuilding = true;
 
diff --git a/pkgs/desktops/gnome-2/desktop/mail-notification/patches/mail-notification-5.4-remove-ubuntu-special-case.patch b/pkgs/desktops/gnome-2/desktop/mail-notification/patches/mail-notification-5.4-remove-ubuntu-special-case.patch
deleted file mode 100644
index 4516f32c6f..0000000000
--- a/pkgs/desktops/gnome-2/desktop/mail-notification/patches/mail-notification-5.4-remove-ubuntu-special-case.patch
+++ /dev/null
@@ -1,33 +0,0 @@
---- jbsrc/lib/src/core/jb-feature.c.~1~	2008-04-27 16:47:27.000000000 +0200
-+++ jbsrc/lib/src/core/jb-feature.c	2008-07-22 11:40:50.856886210 +0200
-@@ -164,8 +164,6 @@
- static void
- gconf_configure (void)
- {
--  JBVariable *variable;
--
-   jb_require_program("gconftool-2");
- 
-   if (! strcmp(jb_variable_get_string("gconf-config-source"), "autodetect"))
-@@ -178,21 +176,6 @@
-       jb_variable_set_string("gconf-config-source", config_source);
-       g_free(config_source);
-     }
--
--  /* fix the default schemas dir on Ubuntu */
--  variable = jb_variable_get_variable_or_error("gconf-schemas-dir");
--  if (! variable->user_set)
--    {
--      static const char *ubuntu_dir = "$datadir/gconf/schemas";
--      char *expanded;
--
--      expanded = jb_variable_expand(ubuntu_dir, NULL);
--
--      if (g_file_test(expanded, G_FILE_TEST_IS_DIR))
--	jb_variable_set_string("gconf-schemas-dir", ubuntu_dir);
--
--      g_free(expanded);
--    }
- }
- 
- static void
```

**Pull request merged**